### ISSUE TITLE
datasource-toolkit/type validation ArrayOfJson causing issues

### ISSUE THREAD

**Issue by ajubin** [ForestAdmin/agent-nodejs#599]
## Expected behavior

After the fix of issue #577 , I still got an error.

## Failure Logs

When updating a JSON field with `["some@email.come"]` it's not recognized as Json.

Here's a dirty fix I used

```
iff --git a/node_modules/@forestadmin/datasource-toolkit/dist/src/validation/field.js b/node_modules/@forestadmin/datasource-toolkit/dist/src/validation/field.js
index 74af6e8..8b26b39 100644
--- a/node_modules/@forestadmin/datasource-toolkit/dist/src/validation/field.js
+++ b/node_modules/@forestadmin/datasource-toolkit/dist/src/validation/field.js
@@ -47,6 +47,13 @@ class FieldValidator {
             return;
         }
         const type = type_getter_1.default.get(value, schema.columnType);
+        
+        // We encouter a bug here, the JSON we entered was an array ["email1"] but was not allowed even if it was valid json
+        // Issue here https://github.com/ForestAdmin/agent-nodejs/issues/577
+        if (type === 'ArrayOfJson' && allowedTypes.includes('Json') ) {
+            allowedTypes.push('ArrayOfJson')
+        }
+
         if (allowedTypes && !allowedTypes.includes(type)) {
             throw new errors_1.ValidationError(`Wrong type for "${field}": ${value}. Expects ${allowedTypes}`);
         }

```

## Context

- Package Version: @forestadmin/datasource-toolkit@1.2.0

**Comment by Scra3** [comment]
Hello and thanks for the report.
Unfortunately I can't reproduce your bug in the version 1.2.0. The last fix should fix your issue. Do you have update your dependency ? 

The merged test to reproduce your case: 

```
test('valid (json array) value type should not throw error', () => {
    expect(() =>
      FieldValidator.validateValue(
        'json',
        factories.columnSchema.build({
          columnType: 'Json',
        }),
        ['email'],
      ),
    ).not.toThrow();
  });
```

In case, you have update your dependency, can you share more details to help me to reproduce please ? 🙏

**Comment by ajubin** [comment]
Hi, 

I double checked by removing and reinstalling my node_modules, I'm running version  `"@forestadmin/datasource-toolkit" "1.2.0"`

I've still got the following error

```

 Wrong type for "jobApplicationRecipientsEmails": email1,email2. Expects Json,Null 

ValidationError: Wrong type for "jobApplicationRecipientsEmails": email1,email2. Expects Json,Null
    at FieldValidator.validateValue (/Users/ajubin/Documents/muzzo/job-marketplace-admin/node_modules/@forestadmin/datasource-toolkit/dist/src/validation/field.js:51:19)
    at /Users/ajubin/Documents/muzzo/job-marketplace-admin/node_modules/@forestadmin/datasource-toolkit/dist/src/validation/field.js:23:56
    at Array.forEach (<anonymous>)
    at FieldValidator.validate (/Users/ajubin/Documents/muzzo/job-marketplace-admin/node_modules/@forestadmin/datasource-toolkit/dist/src/validation/field.js:23:24)
    at RecordValidator.validate (/Users/ajubin/Documents/muzzo/job-marketplace-admin/node_modules/@forestadmin/datasource-toolkit/dist/src/validation/record.js:19:33)
    at UpdateRoute.handleUpdate (/Users/ajubin/Documents/muzzo/job-marketplace-admin/node_modules/@forestadmin/agent/dist/routes/modification/update.js:22:46)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async ErrorHandling.errorHandler (/Users/ajubin/Documents/muzzo/job-marketplace-admin/node_modules/@forestadmin/agent/dist/routes/system/error-handling.js:20:13)
    at async Logger.logger (/Users/ajubin/Documents/muzzo/job-marketplace-admin/node_modules/@forestadmin/agent/dist/routes/system/logger.js:19:13)
    at async bodyParser (/Users/ajubin/Documents/muzzo/job-marketplace-admin/node_modules/koa-bodyparser/index.js:95:5)
```


Here's my field in  `forestadmin-schema.json` 


```
      {
        "defaultValue": null,
        "enums": null,
        "field": "jobApplicationRecipientsEmails",
        "integration": null,
        "inverseOf": null,
        "isFilterable": false,
        "isPrimaryKey": false,
        "isReadOnly": false,
        "isRequired": false,
        "isSortable": true,
        "isVirtual": false,
        "reference": null,
        "type": "Json",
        "validations": []
      },
```

**Comment by Scra3** [comment]
It was my bad, I can reproduce. I will create a ticket and add a fix very soon. 🙏

**Comment by Scra3** [comment]
Hello,
We have merged a [fix](https://github.com/ForestAdmin/agent-nodejs/commit/0ebe1e35c87dfae11043e650856d97daa1ead545) at the release 1.6.3. 
Please tell me if the patch works 🙏

**Comment by romain-gilliotte** [comment]
Closing this as it was fixed by #601
